### PR TITLE
Agnostic namespace test for integration/e2e

### PR DESCRIPTION
**Pull request by onyiny-ang**
**What this PR does / why we need it:**
This is a follow-up test to #159 that:
Allows for integration as well as e2e tests to ensure replicasets created in a namespace are deleted when a namespace is deleted

**Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):**
Not aware of any

Special notes for your reviewer:
namespace tests are failing due to timeout waiting for namespace deletion. Increasing time in "waitForNamespaceDeletion" function (test/common/namespace.go) does not seem to help.

### CODE REVIEW

File: test/common/util.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package common
+
+// TestLogger defines operations common across different types of testing
+type TestLogger interface {
```
Comment (marun) [inline test/common/util.go]: Were you intending to move this from ``pkg/federatedtypes/crudtester`` or have it exist in both places?

File: test/e2e/crud.go [hunk 2]
```diff
@@ -23,15 +23,29 @@ import (
...
+	"k8s.io/federation/test/common"
 	fedframework "k8s.io/federation/test/e2e/framework"
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+var _ = framework.KubeDescribe("Federated namespace [Feature:Federation][Experimental]", func() {
+	f := fedframework.NewDefaultFederatedFramework("federated-namespace")
+	Describe(fmt.Sprintf("Federated namespace resources"), func() {
+		It("should create, read, update and delete successfully", func() {
+			// Load clients only if not skipping to avoid doing
```
Comment (marun) [inline test/e2e/crud.go]: This comment appears unnecessary since there is no conditional skip invocation.  Now that federation is out of tree, skipping likely doesn't make sense anyway since any e2e run in k/f is a federation test run.

File: test/e2e/crud.go [hunk 3]
```diff
@@ -23,15 +23,29 @@ import (
...
+			common.CheckNamespaceContentsRemoved(clientset, logger)
+		})
+	})
+})
+
 var _ = framework.KubeDescribe("Federated types [Feature:Federation][Experimental] ", func() {
 	var clusterClients []kubeclientset.Interface
 
 	f := fedframework.NewDefaultFederatedFramework("federated-types")
-
```
Comment (marun) [inline test/e2e/crud.go]: Unrelated formatting change?

**Review by marun — COMMENTED** [review]
Some nits inline.  Will see if I can troubleshoot the deletion check failure you're seeing.

File: test/common/util.go (hunk 1 above)
Comment (onyiny-ang) [reply test/common/util.go]: I was intending to move it but if I remember correctly, when I tried, there were other completely unrelated things that depended on it as well and I didn't want to break them. I will check this again.

File: test/e2e/crud.go (hunk 2 above)
Comment (onyiny-ang) [reply test/e2e/crud.go]: Ok, I will remove it from the Federated types method (as well as the conditional it references) as well then.

File: image/Makefile [hunk 4]
```diff
@@ -15,7 +15,7 @@
 # Build the fcp image.
 #
 # Usage:
-#   [ARCH=amd64] [REGISTRY="gcr.io/google-containers"] make (build|push) 
+#   [ARCH=amd64] [REGISTRY="gcr.io/google-containers"] make (build|push)
```
Comment (marun) [inline image/Makefile]: Unrelated formatting change

File: test/e2e/crud.go [hunk 5]
```diff
@@ -39,12 +51,6 @@ var _ = framework.KubeDescribe("Federated types [Feature:Federation][Experimenta
 			It("should be created, read, updated and deleted successfully", func() {
 				fedframework.SkipUnlessFederated(f.ClientSet)
 
-				// Load clients only if not skipping to avoid doing
-				// unnecessary work.  Assume clients can be shared
-				// across tests.
-				if clusterClients == nil {
-					clusterClients = f.GetClusterClients()
```
Comment (marun) [inline test/e2e/crud.go]: Consider reverting this change.  Removing the previous skip logic from the e2e tests is orthogonal to the purpose of this PR.

File: test/e2e/crud.go [hunk 6]
```diff
@@ -23,10 +23,22 @@ import (
...
 	"k8s.io/federation/pkg/federatedtypes"
+	"k8s.io/federation/test/common"
 	fedframework "k8s.io/federation/test/e2e/framework"
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+var _ = framework.KubeDescribe("Federated namespace [Feature:Federation][Experimental]", func() {
+	f := fedframework.NewDefaultFederatedFramework("federated-namespace")
+	Describe(fmt.Sprintf("Federated namespace resources"), func() {
+		It("should create, read, update and delete successfully", func() {
```
Comment (marun) [inline test/e2e/crud.go]: Consider updating this test description to reflect the nature of this test (can likely cribbed from the test that is being replaced).

File: test/e2e/crud.go [hunk 7]
```diff
@@ -23,10 +23,22 @@ import (
 
 	kubeclientset "k8s.io/client-go/kubernetes"
 	"k8s.io/federation/pkg/federatedtypes"
+	"k8s.io/federation/test/common"
 	fedframework "k8s.io/federation/test/e2e/framework"
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+var _ = framework.KubeDescribe("Federated namespace [Feature:Federation][Experimental]", func() {
```
Comment (marun) [inline test/e2e/crud.go]: I think there should be a single top-level ``var _ = framework.KubeDescribe(...)`` in a given file with multiple ``Describe`` calls within (with ``test/e2e/namespace.go`` being an example of this structure).

Does this new test belong in ``crud.go`` or would it make more sense to replace the existing test in ``namespace.go``?  If the former, the PR needs to ensure the removal of the existing test.

**Review by marun — COMMENTED** [review]
Some more nits.

**Comment by marun** [comment]
Tests pass locally for me.  Let's see what ci says.

File: test/e2e/crud.go (hunk 7 above)
Comment (onyiny-ang) [reply test/e2e/crud.go]: It probably makes more sense to replace the namespace.go test

File: test/common/util.go (hunk 1 above)
Comment (onyiny-ang) [reply test/common/util.go]: responded to this in latest commit
